esmvalcore/preprocessor/__init__.py

```diff
@@ -461,13 +463,35 @@ def is_closed(self):
 
     def _initialize_entity(self):
         """Initialize the entity representing the file."""
-        super(PreprocessorFile, self)._initialize_entity()
+        super()._initialize_entity()
         settings = {
             'preprocessor:' + k: str(v)
             for k, v in self.settings.items()
         }
         self.entity.add_attributes(settings)
 
+    def group(self, keys: list) -> str:
```

Comment:
Why do think it should be private?

Reply:
it don't have to be, but anything that's not private kind of needs proper numpy-style documentation

Reply:
All methods in this class are documented in the same way, why is this an issue for `group` but not for `check`,  `apply` and other?

Reply:
well, if some old methods are poorly documented it don't mean new ones should be too :grin: It's fine, don't worry, I think in the interest of time we should leave it like this - I really like this feature and it would be awesome if we pushed it to v2.5